docs/developer-guide/working-with-plugins.md

```diff
@@ -66,6 +66,28 @@ module.exports = {
 };
 ```
 
+### Custom Environments in Plugins
+
+Plugins can also provide custom environments. Custom environments can declare `globals`, turn `ecmaFeatures` on and off,
+and configure both core rules as well as rules included in the plugin. To provide custom environment, include `environment`
+property in your exported object.
+
+```js
+module.exports = {
+    environments: {
+        "my-custom-environment": {
+            globals: {
+                "global-variable": true
+            },
+            rules: {
+                "no-alert": 0, //turn off no-alert rule
+                "my-plugin/my-rule": 1 //turn on custom rule from this plugin (given that plugin's name is "my-plugin")
```

Comment:
I was thinking about it as well, and went through the code to find the right place to insert that logic. That's when I figured out that eslint.js itself never parses rule names (and opened the issue about missing documentation).
On the other hand, I'm not sure I really like `*/my-rule` way. Currently there're two ways to point to the rule. It's always either `plugin-name/my-rule` or `my-rule`. This adds a third way, which might just add to the confusion.